dask_planner/src/expression.rs

```diff
@@ -203,15 +225,44 @@ impl PyExpr {
                 Ok(operands)
             }
             Expr::Cast { expr, data_type: _ } => {
+                Ok(vec![PyExpr::from(*expr.clone(), self.input_plan.clone())])
+            }
+            Expr::Case {
+                expr,
+                when_then_expr,
+                else_expr,
+            } => {
                 let mut operands: Vec<PyExpr> = Vec::new();
-                let ex: Expr = *expr.clone();
-                let py_ex: PyExpr = PyExpr::from(ex, self.input_plan.clone());
-                operands.push(py_ex);
+
+                if let Some(e) = expr {
+                    operands.push(PyExpr::from(*e.clone(), self.input_plan.clone()));
+                };
+
+                for (when, then) in when_then_expr {
+                    operands.push(PyExpr::from(*when.clone(), self.input_plan.clone()));
+                    operands.push(PyExpr::from(*then.clone(), self.input_plan.clone()));
+                }
+
+                if let Some(e) = else_expr {
+                    operands.push(PyExpr::from(*e.clone(), self.input_plan.clone()));
+                };
+
                 Ok(operands)
             }
-            _ => Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(
-                "unknown Expr type encountered",
-            )),
+            Expr::Between {
+                expr,
+                negated: _,
```

Comment:
We should raise a followup issue to handle `not between` with test cases that probably make use of the `negated` attribute

Reply:
Done, https://github.com/dask-contrib/dask-sql/issues/519